certbot-apache/certbot_apache/http_01.py

```diff
@@ -89,15 +89,28 @@ def prepare_http01_modules(self):
                     self.configurator.enable_mod(mod, temp=True)
 
     def _mod_config(self):
+        selected_vhosts = []  # type: List[VirtualHost]
+        http_port = str(self.configurator.config.http01_port)
         for chall in self.achalls:
-            vh = self.configurator.find_best_http_vhost(
```

Comment:
One change I think we're making by no longer using this function is we no longer match vhosts that contain the domain in the vhost definition. See https://github.com/certbot/certbot/blob/9671985885ed8106bf88c1c6250ed3a9b010a094/certbot-apache/certbot_apache/configurator.py#L654

I think this fine because I think hardly anyone does this and for people who do, we now match all HTTP vhosts, however, I'd appreciate you double checking my thinking here.

Reply:
You are right. In addition to us mitigating this by configuring all HTTP vhosts, the name in vhost definition is mainly used for binding to a specific `IP:PORT`, so this can actually give Certbot a false positive if the DNS record points to another `IP:PORT` handled by the host.